terracotta/drivers/mysql.py

```diff
@@ -92,22 +27,12 @@ class MySQLDriver(RasterDriver):
 
     This driver caches raster data and key names, but not metadata.
     """
-    _MAX_PRIMARY_KEY_LENGTH = 767 // 4  # Max key length for MySQL is at least 767B
-    _METADATA_COLUMNS: Tuple[Tuple[str, ...], ...] = (
-        ('bounds_north', 'REAL'),
-        ('bounds_east', 'REAL'),
-        ('bounds_south', 'REAL'),
-        ('bounds_west', 'REAL'),
-        ('convex_hull', 'LONGTEXT'),
-        ('valid_percentage', 'REAL'),
-        ('min', 'REAL'),
-        ('max', 'REAL'),
-        ('mean', 'REAL'),
-        ('stdev', 'REAL'),
-        ('percentiles', 'BLOB'),
-        ('metadata', 'LONGTEXT')
-    )
-    _CHARSET: str = 'utf8mb4'
```

Comment:
It appears not. Running
```python
import terracotta
db = terracotta.get_driver('root:password@localhost/tc_test', provider='mysql')
db.create(['a', 'b'])
```
and then opening the `mysql` database and querying
```SQL
SELECT table_name, column_name, character_set_name FROM information_schema.`COLUMNS` WHERE table_schema = "tc_test";
```
yields
```
+------------+------------------+--------------------+
| TABLE_NAME | COLUMN_NAME      | CHARACTER_SET_NAME |
+------------+------------------+--------------------+
| datasets   | a                | utf8mb4            |
| datasets   | b                | utf8mb4            |
| datasets   | filepath         | utf8mb4            |
| key_names  | description      | utf8mb4            |
| key_names  | index            | NULL               |
| key_names  | key_name         | utf8mb4            |
| metadata   | a                | utf8mb4            |
| metadata   | b                | utf8mb4            |
| metadata   | bounds_east      | NULL               |
| metadata   | bounds_north     | NULL               |
| metadata   | bounds_south     | NULL               |
| metadata   | bounds_west      | NULL               |
| metadata   | convex_hull      | utf8mb4            |
| metadata   | max              | NULL               |
| metadata   | mean             | NULL               |
| metadata   | metadata         | utf8mb4            |
| metadata   | min              | NULL               |
| metadata   | percentiles      | NULL               |
| metadata   | stdev            | NULL               |
| metadata   | valid_percentage | NULL               |
| terracotta | version          | utf8mb4            |
+------------+------------------+--------------------+
```

However, I don't know why it is not (any longer?) necessary, nor whether it might be a good defensive strategy to use it.

Reply:
Pretty sure this is MySQL vendor / version dependent. Better supply it to be safe.